### PR TITLE
TIP-236: Clean the user bundle/component

### PR DESCRIPTION
**Pull request by aRn0D**
**Description**

Related PR https://github.com/akeneo/pim-enterprise-dev/pull/3651

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/UserBundle/Security/UserProvider.php [hunk 1]
```diff
@@ -2,81 +2,66 @@
 
 namespace Pim\Bundle\UserBundle\Security;
 
-use Pim\Bundle\UserBundle\Entity\UserInterface;
```
Comment (skeleton) [inline src/Pim/Bundle/UserBundle/Security/UserProvider.php]: you can remove this class and use the vendor/symfony/symfony/src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php imo

File: src/Pim/Bundle/UserBundle/Doctrine/ORM/Repository/GroupRepository.php [hunk 2]
```diff
@@ -47,20 +67,19 @@ public function getAllButDefaultQB()
...
+     *
+     * @return QueryBuilder
      */
-    public function getIdentifierProperties()
+    public function getUserQueryBuilder(GroupInterface $group)
     {
-        return ['name'];
+        return $this->_em->createQueryBuilder()
+            ->select('u')
+            ->from(UserInterface::, 'u')
```
Comment (skeleton) [inline src/Pim/Bundle/UserBundle/Doctrine/ORM/Repository/GroupRepository.php]: ::class

File: src/Pim/Component/User/Model/GroupInterface.php [hunk 3]
```diff
@@ -2,11 +2,81 @@
...
+    public function getName();
+
+    /**
+     * @param  string $name
+     *
+     * @return Group
+     */
+    public function setName($name);
+
+    public function getRoleLabelsAsString();
```
Comment (skeleton) [inline src/Pim/Component/User/Model/GroupInterface.php]: phpdoc

File: src/Pim/Bundle/UserBundle/Security/UserProvider.php (hunk 1 above)
Comment (aRn0D) [reply src/Pim/Bundle/UserBundle/Security/UserProvider.php]: I prefer this way https://github.com/akeneo/pim-community-dev/pull/7759

File: src/Pim/Bundle/UserBundle/Doctrine/ORM/Repository/GroupRepository.php (hunk 2 above)
Comment (aRn0D) [reply src/Pim/Bundle/UserBundle/Doctrine/ORM/Repository/GroupRepository.php]: :+1:

**Comment by aRn0D** [comment]
I did not add any method, just move some code. I won't add type hint it is out of the scope of the card. The team in charge of this bundle will do that.

**Comment by aRn0D** [comment]
Close in favor of https://github.com/akeneo/pim-community-dev/pull/7769

**Pull request closed without merging**